### ISSUE TITLE
config.py._consider_importhook assumes SOURCES.txt uses platform-specific separator while it is not

### ISSUE THREAD

**Issue by throwable-one** [pytest-dev/pytest#2591]
From here: https://svn.python.org/projects/sandbox/trunk/setuptools/doc/formats.txt

> The filenames always use ``/`` as a path separator, which must be
>   converted back to a platform-specific path whenever they are read.

My ``SOURCES.txt`` has ``/`` even on Windows.

Now loo0k at ``_pytest/config.py:1034``
```
 metadata_files = 'RECORD', 'SOURCES.txt'

        package_files = (
            entry.split(',')[0]
            for entrypoint in pkg_resources.iter_entry_points('pytest11')
            for metadata in metadata_files
            for entry in entrypoint.dist._get_metadata(metadata)
        )

        for fn in package_files:
            is_simple_module = os.sep not in fn and fn.endswith('.py') # LOOK HERE !!!!
            is_package = fn.count(os.sep) == 1 and fn.endswith('__init__.py')
            if is_simple_module:
                module_name, ext = os.path.splitext(fn)
                hook.mark_rewrite(module_name)
            elif is_package:
                package_name = os.path.dirname(fn)
                hook.mark_rewrite(package_name)
```

``os.sep`` should not be used here. It leads to errors like one module is considered to be ``is_simple_module `` on **nix, but not on Windows.

**Comment by throwable-one** [comment]
![bad](https://user-images.githubusercontent.com/7673963/28371907-f0bdc90e-6ca6-11e7-9f16-8f1954eb8ea6.png)
Take a look

**Comment by nicoddemus** [comment]
Indeed, thanks for pointing that out. Would you like to open a PR with this fix?

**Comment by srinivasreddy** [comment]
I have raised a PR here - https://github.com/pytest-dev/pytest/pull/2630

**Comment by nicoddemus** [comment]
Fixed by #2630
